File: Permutation.c
#include <stdio.h>
#include <math.h>

int main() {
    int n, r;
    int result;

    // Taking input
    printf("Enter number of items (n): ");
    scanf("%d", &n);

    printf("Enter number of positions (r): ");
    scanf("%d", &r);

    // Permutation with repetition formula n^r
    result = pow(n, r);

    printf("Permutation with repetition = %d\n", result);

    return 0;
}

File: Permutation.java
import java.util.Scanner;

public class PermutationRepetition {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of items (n): ");
        int n = sc.nextInt();

        System.out.print("Enter number of positions (r): ");
        int r = sc.nextInt();

        int result = (int) Math.pow(n, r);

        System.out.println("Permutation with repetition = " + result);
    }
}
